web/modules/custom/joinup_subscription/src/Form/SubscriptionDashboardForm.php

```diff
@@ -85,6 +86,15 @@ public function buildForm(array $form, FormStateInterface $form_state, AccountIn
     $memberships = $this->relationManager->getUserGroupMembershipsByBundle($user, 'rdf_entity', 'collection');
     $bundle_info = $this->entityTypeBundleInfo->getBundleInfo('node');
 
+    $form['usubscribe_all'] = [
+      '#type' => 'link',
```

Comment:
Here's a **UX point of view** about this approach: Allowing the user to _Unsubscribe from all_ when he has no subscription looks like a bad UX. I see a better way:

* Don't show this button (or show it as a disabled button - hm, it's a link!) if the use has no subscription. You can enclose this element into an `if (...)` or set the `#access` property.
* Add the same logic in `UnsubscribeFromAllCollectionsForm::access()`, so a user cannot hit the "unsubscribe all" confirmation form even by accident.
* You can extend the `joinup_subscription.subscription` service (and interface) with a new method `userHasSubscriptions()` and that can be used in both places. That would be the actual `UnsubscribeFromAllCollectionsForm::getUserMembershipIds()`